src/main/java/org/opensearch/flowframework/util/ParseUtils.java

```diff
@@ -229,6 +247,218 @@ public static User getUserContext(Client client) {
         return User.parse(userStr);
     }
 
+    /**
+     * Add user backend roles filter to search source builder=
+     * @param user the user
+     * @param searchSourceBuilder search builder
+     * @return search builder with filter added
+     */
+    public static SearchSourceBuilder addUserBackendRolesFilter(User user, SearchSourceBuilder searchSourceBuilder) {
+        if (user == null) {
+            return searchSourceBuilder;
+        }
+        BoolQueryBuilder boolQueryBuilder = new BoolQueryBuilder();
+        String userFieldName = "user";
+        String userBackendRoleFieldName = "user.backend_roles.keyword";
+        List<String> backendRoles = user.getBackendRoles() != null ? user.getBackendRoles() : ImmutableList.of();
+        // For normal case, user should have backend roles.
+        TermsQueryBuilder userRolesFilterQuery = QueryBuilders.termsQuery(userBackendRoleFieldName, backendRoles);
+        NestedQueryBuilder nestedQueryBuilder = new NestedQueryBuilder(userFieldName, userRolesFilterQuery, ScoreMode.None);
+        boolQueryBuilder.must(nestedQueryBuilder);
+        QueryBuilder query = searchSourceBuilder.query();
+        if (query == null) {
+            searchSourceBuilder.query(boolQueryBuilder);
+        } else if (query instanceof BoolQueryBuilder) {
+            ((BoolQueryBuilder) query).filter(boolQueryBuilder);
+        } else {
+            // e.g., wild card query
+            throw new FlowFrameworkException("Search API does not support queries other than BoolQuery", RestStatus.BAD_REQUEST);
+        }
+        return searchSourceBuilder;
+    }
+
+    /**
+     * Resolve user and execute the function
+     * @param requestedUser the user to execute the request
+     * @param workflowId workflow id
+     * @param filterByEnabled filter by enabled setting
+     * @param listener action listener
+     * @param function workflow function
+     * @param client node client
+     * @param clusterService cluster service
+     * @param xContentRegistry contentRegister to parse get response
+     */
+    public static void resolveUserAndExecute(
+        User requestedUser,
+        String workflowId,
+        boolean filterByEnabled,
```

Comment:
This unboxes the `Boolean` which we're pretty sure will never be null, otherwise would NPE here.  But we may as well just keep it boxed here to save a few nanoseconds.